src/devices/bus/rs232/xvd701.cpp

```diff
@@ -110,39 +155,192 @@ void jvc_xvd701_device::rcv_complete()
 
 	if (m_command[0] == 0xff &&
 		m_command[1] == 0xff &&
-		m_command[2] == 0x21 &&
-		sum(m_command, sizeof(m_command)) == 0)
+		m_command[10] == sum(m_command, sizeof(m_command) - 1))
 	{
-		// printf("xvd701");
-
-		//for (int i = 0; i < sizeof(m_command); i++)
-		//  printf(" %02x", m_command[i]);
-
-		//printf("\n");
-
-		// FF FF 21 3E 40 70 00 00 00 00 73 DEVICE ON
-		// FF FF 21 3E 40 60 00 00 00 00 03 DEVICE OFF
-		// FF FF 21 0C 44 60 00 00 00 00 31 STOP
-		// FF FF 21 0C 43 75 00 00 00 00 1D PLAY
-		// FF FF 21 0C 43 6D 00 00 00 00 25 PAUSE
-		// FF FF 21 0C 50 20 00 00 00 00 63 SEEK TO SPECIFIC CHAPTER
-		// FF FF 21 0C 50 73 00 00 00 00 12 FF (SEEK TO NEXT CHAPTER)
-		// FF FF 21 0C 50 61 00 00 00 00 24 PREV (SEEK TO PREVIOUS CHAPTER)
-
-		m_response[0] = 0xff;
-		m_response[1] = 0xfe;
-		m_response[2] = 0x7f;
-		m_response[3] = 0x7e;
-		m_response[4] = 0x7d;
-		m_response[5] = 0x7c;
-		m_response[6] = 0x7b;
-		m_response[7] = 0x7a;
-		m_response[8] = 0x79;
-		m_response[9] = 0x78;
-		m_response[10] = 0x77;
-		m_response_index = 0;
-
-		m_timer_response->adjust(attotime::from_msec(100));
+		if (m_command[3] == 0x0c)
+		{
+			// Media packets
+
+			// TODO: 0x41 Drive commands
+
+			// 0x43 Playback commands
+			if (m_command[4] == 0x43 && m_command[5] == 0x6d)
+			{
+				// FF FF 21 0C 43 6D 00 00 00 00 25 PAUSE
+				LOGCMD("xvd701: Playback PAUSE\n");
+				m_playback_status = STATUS_PAUSE;
+				create_packet(STATUS_OK, NO_RESPONSE);
+			}
+			else if (m_command[4] == 0x43 && m_command[5] == 0x75)
+			{
+				// FF FF 21 0C 43 75 00 00 00 00 1D PLAY
+				LOGCMD("xvd701: Playback PLAY\n");
+
+				auto status = STATUS_OK;
+				if (m_playback_status == STATUS_STOP)
+				{
+					// Force video to load again if the video was stopped then started again
+					if (!seek_chapter(m_chapter))
+						status = STATUS_ERROR;
+				}
+
+				if (status == STATUS_OK)
+					m_playback_status = STATUS_PLAYING;
+
+				create_packet(status, NO_RESPONSE);
+			}
+
+			// 0x44 Stop commands
+			else if (m_command[4] == 0x44 && m_command[5] == 0x60)
+			{
+				// FF FF 21 0C 44 60 00 00 00 00 31 STOP
+				LOGCMD("xvd701: Playback STOP\n");
+
+				m_playback_status = STATUS_STOP;
+				create_packet(STATUS_OK, NO_RESPONSE);
+			}
+
+			// TODO: 0x4c Disk parameter commands
+
+			// TODO: 0x4e Disk status commands
+
+			// 0x50 Seek commands
+			else if (m_command[4] == 0x50 && m_command[5] == 0x20)
```

Comment:
Please don’t put blank lines and comments between the end of an `if` controlled statement and a following `else` – it makes it far easier for a person debugging the code to get confused if the bottom of their window shows one or more lines following the end of the controlled statement but the following `else` is scrolled off the bottom.

Reply:
I removed the less useful comments and did away with the nesting/attempted groupings because even those felt somewhat forced or not helpful at times. The code should be easier to follow at a glance now.